StyleCop.Analyzers/StyleCop.Analyzers.Test/OrderingRules/SA1203UnitTests.cs

```diff
@@ -145,11 +169,126 @@ public class Test
+
+            var fixedTestCode = @"
+public class Test
+{
+    const int Test2 = 2;
+    const int Test3 = 3;
+    private int Test1 = 1;
+}";
+            await this.VerifyCSharpDiagnosticAsync(fixedTestCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix will move the non constant fields before the constant ones, only using a single access modifier.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestCodeFixSingleAccessModifierAsync()
+        {
+            var testCode = @"
+public class Foo
+{
+    public const string Before1 = ""test"";
+
+    public const string Before2 = ""test"";
+
+    public int field1;
+
+    public const string After1 = ""test"";
+
+    public int between;
+
+    public const string After2 = ""test"";
+}
+";
+
+            var fixedTestCode = @"
+public class Foo
+{
+    public const string Before1 = ""test"";
+
+    public const string Before2 = ""test"";
+
+    public const string After1 = ""test"";
+
+    public const string After2 = ""test"";
+
+    public int field1;
+
+    public int between;
+}
+";
+
+            var diagnosticResults = new[]
+            {
+                this.CSharpDiagnostic().WithLocation(10, 25).WithArguments("public"),
+                this.CSharpDiagnostic().WithLocation(14, 25).WithArguments("public")
+            };
+            await this.VerifyCSharpDiagnosticAsync(testCode, diagnosticResults, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpDiagnosticAsync(fixedTestCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix will move the non constant fields before the constant ones.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestCodeFixAsync()
+        {
+            var testCode = @"public class Foo
+{
+    private const string Before1 = ""test"";
+
+    public const string Before2 = ""test"";
+
+    private int field1;
+
+    private const string After1 = ""test"";
+
+    public int between;
+
+    public const string After2 = ""test"";
+}
+";
+
+            var fixedTestCode = @"public class Foo
+{
+
+    public const string Before2 = ""test"";
+
+    public const string After2 = ""test"";
+
+    public int between;
+    private const string Before1 = ""test"";
+
+    private const string After1 = ""test"";
+
+    private int field1;
```

Comment:
While I like the final order, I'm thinking we should only move the fields that have diagnostics reported. I propose the resulting order would be this.

```
private const string Before1 = "test";
public const string Before2 = "test";
public const string After2 = "test"; // moved up until no longer in violation
private const string After1 = "test"; // moved up until no longer in violation
private int field1;
public int between;
```

@sharwell @hvanbakel What do you think of this? I believe that fixing the other violations should be handled when fixing SA1202 violations and are out of scope for SA1203.